python/tank/deploy/tank_commands/validate_config.py

```diff
@@ -46,36 +103,112 @@ def run_interactive(self, log, args):
+
+        self._interactive = True
+
+        parser = argparse.ArgumentParser(
+            prog="./tank validate",
+            description="Environment validation command."
+        )
+
+        # a list of environment names
+        parser.add_argument(
+            "envs",
+            metavar="<env_name>",
+            type=str,
+            nargs="*",
+            help=self.parameters["envs"]["description"],
+        )
+
+        # mutually exclusive dump types, full and sparse
+        dump_type_group = parser.add_mutually_exclusive_group()
+        dump_type_group.add_argument(
+            "-f", "--dump",
+            action="store_true",
+            help=self.parameters["dump"]["description"],
+        )
+        dump_type_group.add_argument(
+            "-s", "--dump-sparse",
+            action="store_true",
+            help=self.parameters["dump_sparse"]["description"],
+        )
+
+        # the output directory for the modified environment files
+        parser.add_argument(
+            "-l", "--dump-location",
+            metavar="<output_directory>",
+            type=str,
+            help=self.parameters["dump_sparse"]["description"],
+        )
+
+        # turn on debug comments in the dumped environments
+        parser.add_argument(
+            "--dump-debug",
+            action="store_true",
+            help=self.parameters["dump_debug"]["description"],
+        )
+
+        # calling vars() on the returned namespace object essentially
+        # translates it into a dictionary. In addition, argparse naturally
+        # translates the flags we used to the matching parameters that
+        # are defined by the action (ex: "--dump-debug" becomes "dump_debug").
+        # so we're left with a dict that should look just like the params
+        # supplied to the non-interactive version of this action.
+        parsed_args = vars(parser.parse_args(args))
+
+        # validate the parsed args just like the non-interactive params.
+        computed_params = self.__validate_parameters(parsed_args)
+
+        # do work.
+        return self._run(log, computed_params)
         
-    def _run(self, log):
+    def _run(self, log, params):
         """
-        Actual execution payload
-        """ 
-        
+        Actual execution payload. Includes validation and optional dumping.
+
+        :param log: A logger instance.
+        :param params: A dict of parameters to drive the validation/dumping.
+        :return:
+        """
+
         log.info("")
         log.info("")
         log.info("Welcome to the Shotgun Pipeline Toolkit Configuration validator!")
         log.info("")
     
         try:
-            envs = self.tk.pipeline_configuration.get_environments()
+            env_names = self.tk.pipeline_configuration.get_environments()
         except Exception, e:
-            raise TankError("Could not find any environments for config %s: %s" % (self.tk, e))
+            raise TankError(
```

Comment:
I didn't change the error message, but I can if you like! I shouldn't have modified the line, but my 80 character limit inner demon compelled me to.